contrib/coredns/policy/policy_test.go

```diff
@@ -467,3 +471,60 @@ func TestEdns(t *testing.T) {
 		}
 	}
 }
+
+// dnstap test
+
+type testIORoutine struct {
+	dnstapChan chan dnstap.Dnstap
+}
+
+func newIORoutine() testIORoutine {
+	ch := make(chan dnstap.Dnstap, 3)
+	io := testIORoutine{dnstapChan: ch}
+	// close channel by timeout to prevent checker from waiting forever
+	go func() {
+		time.Sleep(5 * time.Second)
+		close(ch)
+	}()
+	return io
+}
+
+func (io testIORoutine) Dnstap(msg dnstap.Dnstap) {
+	io.dnstapChan <- msg
+}
+
+func TestPolicyWithDnstap(t *testing.T) {
+	tapIO := newIORoutine()
+	pm := PolicyPlugin{Next: handler(), TapIO: tapIO, options: make(map[uint16][]edns0Map)}
+
+	req := new(dns.Msg)
+	req.SetQuestion("test.com.", dns.TypeA)
+
+	permit := &pdp.Response{Effect: pdp.Response_PERMIT}
+	pm.pdp = newTestClientInit(permit, permit, nil, nil)
+
+	trapper := dtest.TrapTapper{Full: true}
+	tapRW := taprw.ResponseWriter{
+		Query:          new(dns.Msg),
+		ResponseWriter: &test.ResponseWriter{},
+		Tapper:         &trapper,
+	}
+	pm.ServeDNS(context.TODO(), &tapRW, req)
```

Comment:
I prefer to have here context.Background because don't expect that you or anyone is going to change it to something else.

Reply:
Thank you. I just copied this call from another test. Would you like me also to change to context.Background in other tests in this file?